### PR TITLE
fix(tests): add a way to disable client metrics stderr

### LINKED ISSUE #3158: client.metrics make for a very verbose Travis log
See https://travis-ci.org/mozilla/fxa-content-server/jobs/84933246

Sample:

```
{"time":"2015-10-12T15:00:00.000Z","op":"client.metrics","hostname":"testing-worker-linux-docker-219dc4c8-3200-linux-8","pid":3615,"v":1,"lang":"en","agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:40.0) Gecko/20100101 Firefox/40.0 FxATester/1.0","duration":5768,"context":"web","broker":"base","entrypoint":"none","service":"none","migration":"none","campaign":"none","nt.navigationStart":0,"nt.unloadEventStart":154,"nt.unloadEventEnd":161,"nt.fetchStart":141,"nt.domainLookupStart":141,"nt.domainLookupEnd":141,"nt.connectStart":141,"nt.connectEnd":141,"nt.requestStart":142,"nt.responseStart":149,"nt.responseEnd":149,"nt.domLoading":154,"nt.domInteractive":254,"nt.domContentLoadedEventStart":259,"nt.domContentLoadedEventEnd":260,"nt.domComplete":272,"nt.loadEventStart":272,"nt.loadEventEnd":272,"events":["signup.submit","signup.success","screen.confirm"],"event_durations":[3983,4185,4212],"screen.width":1024,"screen.height":768}

{"time":"2015-10-12T15:00:00.000Z","op":"client.metrics","hostname":"testing-worker-linux-docker-219dc4c8-3200-linux-8","pid":3615,"v":1,"lang":"en","agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:40.0) Gecko/20100101 Firefox/40.0 FxATester/1.0","duration":3136,"context":"web","broker":"base","entrypoint":"none","service":"none","migration":"none","campaign":"none","nt.navigationStart":0,"nt.unloadEventStart":171,"nt.unloadEventEnd":178,"nt.fetchStart":155,"nt.domainLookupStart":155,"nt.domainLookupEnd":155,"nt.connectStart":155,"nt.connectEnd":155,"nt.requestStart":155,"nt.responseStart":163,"nt.responseEnd":163,"nt.domLoading":171,"nt.domInteractive":248,"nt.domContentLoadedEventStart":255,"nt.domContentLoadedEventEnd":256,"nt.domComplete":264,"nt.loadEventStart":264,"nt.loadEventEnd":264,"events":["screen.signup","signup.email-optin.visible.true","signup.experiment.pw_strength.enabled","loaded","signup.experiment.pw_strength.all_letters_or_numbers"],"event_durations":[1284,1290,1310,1325,3131],"screen.width":1024,"screen.height":768}

✓ firefox on any platform - sign_up - sign up with a verified account forces the user to sign in (4.76s)

{"time":"2015-10-12T15:00:00.000Z","op":"client.metrics","hostname":"testing-worker-linux-docker-219dc4c8-3200-linux-8","pid":3615,"v":1,"lang":"en","agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:40.0) Gecko/20100101 Firefox/40.0 FxATester/1.0","duration":5717,"context":"web","broker":"base","entrypoint":"none","service":"none","migration":"none","campaign":"none","nt.navigationStart":0,"nt.unloadEventStart":171,"nt.unloadEventEnd":178,"nt.fetchStart":155,"nt.domainLookupStart":155,"nt.domainLookupEnd":155,"nt.connectStart":155,"nt.connectEnd":155,"nt.requestStart":155,"nt.responseStart":163,"nt.responseEnd":163,"nt.domLoading":171,"nt.domInteractive":248,"nt.domContentLoadedEventStart":255,"nt.domContentLoadedEventEnd":256,"nt.domComplete":264,"nt.loadEventStart":264,"nt.loadEventEnd":264,"events":["signup.submit","error.signup.auth.101","screen.signin","signin.ask-password.shown.account-unknown","signin.profile_image_not_shown"],"event_durations":[3446,3526,4173,4175,4194],"screen.width":1024,"screen.height":768}

{"time":"2015-10-12T15:00:00.000Z","op":"client.metrics","hostname":"testing-worker-linux-docker-219dc4c8-3200-linux-8","pid":3615,"v":1,"lang":"en","agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:40.0) Gecko/20100101 Firefox/40.0 FxATester/1.0","duration":3364,"context":"web","broker":"base","entrypoint":"none","service":"none","migration":"none","campaign":"none","nt.navigationStart":0,"nt.unloadEventStart":140,"nt.unloadEventEnd":147,"nt.fetchStart":127,"nt.domainLookupStart":127,"nt.domainLookupEnd":127,"nt.connectStart":127,"nt.connectEnd":127,"nt.requestStart":128,"nt.responseStart":140,"nt.responseEnd":140,"nt.domLoading":140,"nt.domInteractive":229,"nt.domContentLoadedEventStart":237,"nt.domContentLoadedEventEnd":238,"nt.domComplete":250,"nt.loadEventStart":250,"nt.loadEventEnd":250,"events":["screen.signup","signup.email-optin.visible.true","signup.experiment.pw_strength.enabled","loaded","signup.experiment.pw_strength.long_enough"],"event_durations":[1516,1522,1547,1565,3359],"screen.width":1024,"screen.height":768}

✓ firefox on any platform - sign_up - sign up with an unverified account and different password re-signs up user (4.317s)

{"time":"2015-10-12T15:00:00.000Z","op":"client.metrics","hostname":"testing-worker-linux-docker-219dc4c8-3200-linux-8","pid":3615,"v":1,"lang":"en","agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:40.0) Gecko/20100101 Firefox/40.0 FxATester/1.0","duration":5319,"context":"web","broker":"base","entrypoint":"none","service":"none","migration":"none","campaign":"none","nt.navigationStart":0,"nt.unloadEventStart":140,"nt.unloadEventEnd":147,"nt.fetchStart":127,"nt.domainLookupStart":127,"nt.domainLookupEnd":127,"nt.connectStart":127,"nt.connectEnd":127,"nt.requestStart":128,"nt.responseStart":140,"nt.responseEnd":140,"nt.domLoading":140,"nt.domInteractive":229,"nt.domContentLoadedEventStart":237,"nt.domContentLoadedEventEnd":238,"nt.domComplete":250,"nt.loadEventStart":250,"nt.loadEventEnd":250,"events":["signup.submit","signup.success","screen.confirm"],"event_durations":[3691,3888,3906],"screen.width":1024,"screen.height":768}

{"time":"2015-10-12T15:00:00.000Z","op":"client.metrics","hostname":"testing-worker-linux-docker-219dc4c8-3200-linux-8","pid":3615,"v":1,"lang":"en","agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:40.0) Gecko/20100101 Firefox/40.0 FxATester/1.0","duration":3040,"context":"web","broker":"base","entrypoint":"none","service":"none","migration":"none","campaign":"none","nt.navigationStart":0,"nt.unloadEventStart":160,"nt.unloadEventEnd":169,"nt.fetchStart":146,"nt.domainLookupStart":146,"nt.domainLookupEnd":146,"nt.connectStart":146,"nt.connectEnd":146,"nt.requestStart":147,"nt.responseStart":152,"nt.responseEnd":152,"nt.domLoading":160,"nt.domInteractive":252,"nt.domContentLoadedEventStart":252,"nt.domContentLoadedEventEnd":253,"nt.domComplete":271,"nt.loadEventStart":271,"nt.loadEventEnd":271,"events":["screen.signup","signup.email-optin.visible.true","signup.experiment.pw_strength.enabled","loaded","signup.experiment.pw_strength.all_letters_or_numbers"],"event_durations":[1309,1312,1329,1345,3035],"screen.width":1024,"screen.height":768}
```

That's a lot of client metrics info that I don't care about when trying to figure out which test is failing.

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #3158

### CODE REVIEW

**Comment by vladikoff** [comment]
@shane-tomlinson r?

File: server/lib/configuration.js [hunk 1]
```diff
@@ -48,6 +48,13 @@ var conf = module.exports = convict({
     default: path.resolve(__dirname, '..', '..', 'cert.pem'),
     doc: 'The location of the SSL certificate in pem format'
   },
+  client_metrics: {
+    enabled: {
```
Comment (shane-tomlinson) [inline server/lib/configuration.js]: `client_metrics.enabled` implies a global on/off switch of client metrics, whereas this flag only enables/disables output to stderr. Maybe this should be something like:

```js
client_metrics: {
  stderr_collector_enabled: {
     ...
  },
  statsd_collector_enabled: {
    ...
  }
}
```

I could also see something like: 
```js
client_metrics: {
  collectors: {
     default: [ 'ga', 'stderr', 'statsd' ],
     doc: 'Where to send client metrics'
  }
}
```

**Comment by shane-tomlinson** [comment]
> shane-tomlinson was assigned by rfk 4 hours ago

I left my comments, back to @vladikoff

**Comment by vladikoff** [comment]
@shane-tomlinson I updated this to be more consistent with the "DISABLE_..." options we already have. Also added some tests. If we need to mute statsd and other logging we can add that later if required, following the same pattern.

**Comment by vladikoff** [comment]
uh metrics-unit keeps failing on node 0.10, will need to check what is going on, otherwise should be ready.

**Comment by vladikoff** [comment]
Updated and should be ready!

**Comment by shane-tomlinson** [comment]
https://travis-ci.org/mozilla/fxa-content-server/jobs/90769425 MUCH cleaner! Now if we could just get rid of all *this* cruft:

```
[2015-11-12 16:49:54] bid.ccverifier.INFO: compute cluster: spawned new worker process (3882) 1/20 processes running
[2015-11-12 16:49:54] bid.ccverifier.DEBUG: compute cluster: passing compute job to process 3882
[2015-11-12 16:49:54] bid.ccverifier.INFO: compute cluster: spawned new worker process (3884) 2/20 processes running
[2015-11-12 16:49:54] bid.ccverifier.DEBUG: compute cluster: passing compute job to process 3884
[2015-11-12 16:49:55] bid.v2.INFO: assertion_verification_time 334
[2015-11-12 16:49:55] bid.server.INFO: 127.0.0.1 - - [Thu, 12 Nov 2015 16:49:55 GMT] "POST /v2 HTTP/1.1" 200 339 "-" "-"
[2015-11-12 16:49:55] bid.v2.INFO: verify { result: 'success', rp: 'http://127.0.0.1:9010' }
[2015-11-12 16:49:55] bid.ccverifier.DEBUG: compute cluster: process 3882 completed work in 0.35s
[2015-11-12 16:49:55] bid.v2.INFO: assertion_verification_time 344
[2015-11-12 16:49:55] bid.server.INFO: 127.0.0.1 - - [Thu, 12 Nov 2015 16:49:55 GMT] "POST /v2 HTTP/1.1" 200 339 "-" "-"
[2015-11-12 16:49:55] bid.v2.INFO: verify { result: 'success', rp: 'http://127.0.0.1:9010' }
[2015-11-12 16:49:55] bid.ccverifier.DEBUG: compute cluster: process 3884 completed work in 0.33s
[2015-11-12 16:49:55] bid.ccverifier.DEBUG: compute cluster: passing compute job to process 3882
[2015-11-12 16:49:55] bid.v2.INFO: assertion_verification_time 75
[2015-11-12 16:49:55] bid.server.INFO: 127.0.0.1 - - [Thu, 12 Nov 2015 16:49:55 GMT] "POST /v2 HTTP/1.1" 200 339 "-" "-"
[2015-11-12 16:49:55] bid.v2.INFO: verify { result: 'success', rp: 'http://127.0.0.1:9010' }
[2015-11-12 16:49:55] bid.ccverifier.DEBUG: compute cluster: process 3882 completed work in 0.08s
[2015-11-12 16:49:55] bid.ccverifier.DEBUG: compute cluster: passing compute job to process 3882
[2015-11-12 16:49:55] bid.v2.INFO: assertion_verification_time 63
[2015-11-12 16:49:55] bid.server.INFO: 127.0.0.1 - - [Thu, 12 Nov 2015 16:49:55 GMT] "POST /v2 HTTP/1.1" 200 339 "-" "-"
[2015-11-12 16:49:55] bid.v2.INFO: verify { result: 'success', rp: 'http://127.0.0.1:9010' }
[2015-11-12 16:49:55] bid.ccverifier.DEBUG: compute cluster: process 3882 completed work in 0.07s
```

**Pull request merged**